### ISSUE TITLE
AsyncHttpConsumer doesn't have normal consumer attributes

### ISSUE THREAD

**Issue by strexicious** [django/channels#1165]
I was trying to access `self.channel_name` and `self.channel_layer` within a consumer inherited from AsyncHttpConsumer and it raised exceptions saying that my consumer had no such attributes.

Here is the toy code to reproduce this bug:

**routing.py**
```python
application = ProtocolTypeRouter({
    "http": URLRouter([
        url(r"^test/me", MeConsumer),
        url(r"", AsgiHandler)
    ])
})
```
**consumers.py**
```python
class MeConsumer(AsyncHttpConsumer):

    async def handle(self, body):
        my_channel_name = self.channel_name
        self.channel_layer.group_add("my_group", self.channel_name)
        print("Here is the channel name:", my_channel_name)
        await asyncio.sleep(10)
        await self.send_response(200, b"Your response bytes", headers=[
            ("Content-Type", "text/plain"),
        ])
```

*I couldn't help but peek into the source code, and I noticed that `AsyncHttpConsumer` overrides `__call__` of it's base class. I am not an expert but :eyes: it may be useful to know.*

**Comment by matthiask** [comment]
Good catch. That probably also means that the `chat_message` example I added to the docs doesn't work right now since there's no way to dispatch messages through the channel layer to the `AsyncHttpConsumer`. https://channels.readthedocs.io/en/latest/topics/consumers.html#asynchttpconsumer

Maybe the code has to be changed to assemble the request body inside a new `def http_request` method which itself would call `self.handle()` as soon as the request body is complete. Also, we'd probably need a `def http_disconnect` method raising `StopConsumer` or something, and maybe also calling `self.disconnect()`

I'm not working on an ASGI project right now unfortunately, so if someone else wants to pick this up -- feel free. I hope that the pointers above are helpful.

**Comment by andrewgodwin** [comment]
Ah yes, this looks like a bug in the implementation. I'm not going to get to this soon but others are welcome to give it a go.

**Comment by strexicious** [comment]
Sorry for the late response. But seeing that no one has advanced further on this bug I have opened a PR #1172 for the fix. Made a new branch from the 2.1.3 release tag and changed the minimum things, as indicated by @matthiask, to make it work.
Note: first time contributing to any open-source code and I am a really novice programmer. Let me know if you spot any mistakes. Also, since I don't have Python 3.6 (because Ubuntu 16.04), I couldn't run the formatter  (black).

**Comment by JoshYuJump** [comment]
Waiting for a release include this PR #1172

**Comment by andrewgodwin** [comment]
It'll get released when I get time, it might be a few more weeks.
